### PR TITLE
Feature 63 page drafts

### LINKED ISSUE #63: Page drafts for new services and service locations

### PR DESCRIPTION
**Pull request by platonovsimeon**
Resolves #63

### CODE REVIEW

File: page-drafts/page-drafts.php [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  }
+
+  /**
+   * Creates new drafts from new index items
+   * 
+   * @param new_index_items new index items to use
+   * @param template_type template type to use
+   * @param new_sync_time new sync time
+   */
+  function createDrafts($new_index_items, $template_type, $new_sync_time) {
```
Comment (anttileppa) [inline page-drafts/page-drafts.php:201]: snake cases?

File: settings/settings-ui.php [hunk 2]
```diff
@@ -70,13 +78,37 @@ private function addOption($group, $type, $name, $title) {
         ]);
       }
 
-      private function addDropDownOption($group, $type, $name, $title) {
-        add_settings_field($name, $title, [$this, 'createDropDownUI'], SPTV_SETTINGS_PAGE, $group, [
+      private function addVersionDropdown() {
```
Comment (anttileppa) [inline settings/settings-ui.php:81]: Add docs

File: settings/settings-ui.php [hunk 3]
```diff
@@ -70,13 +78,37 @@ private function addOption($group, $type, $name, $title) {
...
+        add_settings_field('version', __( "PTV-versio", 'sptv' ), [$this, 'createVersionDropdownUI'], SPTV_SETTINGS_PAGE, 'elastic', [
+          'name' => 'version', 
+          'type' => 'text'
+        ]);
+      }
+
+      private function addTemplateDropdown($name, $title, $templates) {
```
Comment (anttileppa) [inline settings/settings-ui.php:88]: Add docs

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: page-drafts/page-drafts.php [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+          $schedules['template_interval'] = [
+              'interval' => 900
+          ];
+          return $schedules;
+        });
+      
+        add_action('draft_hook', function () {
+          $lastSyncTime = get_option('sptv-last-template-sync-time');
+          $newSyncTime = gmdate("Y-m-d\TH:i:s\Z");
+          $newIndexItems = $this->getNewIndexItems('v11-service', 'organizationIds', $lastSyncTime);
```
Comment (anttileppa) [inline page-drafts/page-drafts.php:28]: Version should come from the settings

File: page-drafts/page-drafts.php [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+          if (is_array($newIndexItems)) {
+            $this->createDrafts($newIndexItems, 'service-template', $newSyncTime);
+          }
+      
+        });
+      
+        add_action('location_draft_hook', function () {
+          $lastSyncTime = get_option('sptv-last-location-template-sync-time');
+          $newSyncTime = gmdate("Y-m-d\TH:i:s\Z");
+          $newIndexItems = $this->getNewIndexItems('v11-servicelocation-service-channel', 'organizationId', $lastSyncTime);
```
Comment (anttileppa) [inline page-drafts/page-drafts.php:38]: Version should come from the settings

File: settings/settings-ui.php [hunk 6]
```diff
@@ -70,13 +78,47 @@ private function addOption($group, $type, $name, $title) {
...
+          echo "</select >";
+        }, SPTV_SETTINGS_PAGE, 'ptv', [
           'name' => $name, 
-          'type' => $type
+          'type' => 'text'
         ]);
       }
-      public function createDropDownUI($opts) {
+
+      public function createVersionDropDownUI($opts) {
```
Comment (anttileppa) [inline settings/settings-ui.php:121]: Add docs

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: page-drafts/page-drafts.php [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        $login = $username . ':'. $password;
+        $base64Login = base64_encode($login);
+        $requestBody = [
+          'size' => 10000,
+          'query' => [
+            'bool' => [
+              'must' => [
+                [
+                  'term' => [
+                    $organizationFieldName => array_values(Settings::getOrganizationIds())[0]
```
[the remark this answers is not in the document]
Comment (platonovsimeon) [reply page-drafts/page-drafts.php:83]: Fixed

File: page-drafts/page-drafts.php (hunk 4 above)
Comment (platonovsimeon) [reply page-drafts/page-drafts.php:28]: Fixed

File: page-drafts/page-drafts.php (hunk 5 above)
Comment (platonovsimeon) [reply page-drafts/page-drafts.php:38]: Fixed

File: settings/settings-ui.php (hunk 6 above)
Comment (platonovsimeon) [reply settings/settings-ui.php:121]: Fixed

**Pull request merged**